src/init.cpp

```diff
@@ -1795,7 +1795,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     if (WITH_LOCK(chainman.GetMutex(), return chainman.ActiveTip() == nullptr)) {
         WAIT_LOCK(kernel_notifications.m_tip_block_mutex, lock);
         kernel_notifications.m_tip_block_cv.wait(lock, [&]() EXCLUSIVE_LOCKS_REQUIRED(kernel_notifications.m_tip_block_mutex) {
-            return !kernel_notifications.m_tip_block.IsNull() || ShutdownRequested(node);
+            return kernel_notifications.m_tip_block.has_value() || ShutdownRequested(node);
```

Comment:
I've noticed that we're checking whether the optional has a value in different ways here and in `interfaces.cpp`.
If you change the PR again, please consider unifying:
```suggestion
            return kernel_notifications.m_tip_block || ShutdownRequested(node);
```

Reply:
The reason I did it differently was that `interfaces.cpp` the pattern `blah && blah.value()` makes it clear the first value is a pointer rather than a boolean. That's not clear here. That said, I don't think anyone would be confused either.